src/transform/parameters.ts

```diff
@@ -34,6 +36,8 @@ export function transformParametersArray(
         } else if (version === 3) {
           mappedParams[reference.in][reference.name || paramName] = {
             ...reference,
+            // They want to keep this ref here so it references
+            // the components, but it does violate the standards.
```

Comment:
Could you explain a little more about how this violates the standards? Should this be removed?

Reply:
Consider a document that has a reference as for a parameter to #/components/parameters/foobar.

This code requires the schema attribute to be that reference, but it turns out the code that I've written for determining if there is a default value for the parameter expects the schema for the parameter itself to be in the schema property that is getting written here.

So to access the true schema of the parameters the code has to follow .schema.schema, which seems quite awkward to reason about but this PR does implement it.

Reply:
Ah yes for parameters we did try and flatten the schema, because in the TypeScript conversion we’re more interested in types, and we found the interface easier to use without nested `.schema` … `.schema` everywhere. I don’t know if I would say this “violates the standards” because TypeScript types are not bound to the standards. There are some opinions and interpretations we’re making here.

Reply:
I guess in a related question: what’s the purpose for this comment? Is this a TODO? Does this clarify the purpose of the existing code? I’m a little unclear why this was added when no other code here was changed.

Reply:
It was a note to myself, you can merge it or not merge it as you like.  But it may help other users who want to edit this code understand what is going on.